### PR TITLE
Adding recipe: "fmt"

### PR DESCRIPTION
**Pull request by tdegeus**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@conda-forge/help-c-cpp

### CODE REVIEW

**Comment by tdegeus** [comment]
@conda-forge/help-c-cpp 

I'm getting a compiler error on Windows:

```none
fatal error C1083: Cannot open include file: 'cstdint': No such file or directory
```

It seems to be associated with rather old Visual Studio versions. Is this a known issue? Is there a workaround?

**Comment by scopatz** [comment]
Yep,you can always provide a patch in the recipe and/or skip windows for now

**Comment by tdegeus** [comment]
@scopatz Thanks for your response. Skipping Windows builds seems an acceptable first step. I can then patch when a PR is accepted in fmt. That being said... How do I generate patches?

File: recipes/fmt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/fmtlib/fmt/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
```
Comment (isuruf) [inline recipes/fmt/meta.yaml]:
```suggestion
  skip: True  # [win and vc<14]
```

Comment (tdegeus) [reply recipes/fmt/meta.yaml]: @isuruf Why is this needed if `win` is entirely skipped?

**Comment by isuruf** [comment]
Can you build a shared library instead of a static one?

**Comment by tdegeus** [comment]
@isuruf Thanks for your comment. Will adding the cmake option `-DBUILD_SHARED_LIBS:BOOL=TRUE` do the job?

File: recipes/fmt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  commands:
+    - test -d ${PREFIX}/include/fmt  # [unix]
+    - test -f ${PREFIX}/include/fmt/core.h  # [unix]
+    - test -f ${PREFIX}/include/fmt/format.h  # [unix]
+    - test -f ${PREFIX}/lib/cmake/fmt/fmt-config.cmake  # [unix]
+    - if exist %LIBRARY_PREFIX%\include\fmt\core.h (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\include\fmt\format.h (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\lib\cmake\fmt-config.cmake (exit 0) else (exit 1)  # [win]
```
Comment (scopatz) [inline recipes/fmt/meta.yaml]: You probably want to test that the shared library exists here too

File: recipes/fmt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+about:
+  home: https://github.com/tdegeus/pyxtensor
```
Comment (scopatz) [inline recipes/fmt/meta.yaml]: This doesn't seem right :wink:

File: recipes/fmt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
```
Comment (scopatz) [inline recipes/fmt/meta.yaml]: You also need make on linux here

**Comment by tdegeus** [comment]
Thanks @scopatz and @isuruf for your comments!

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**